sklearn/linear_model/tests/test_least_angle.py

```diff
@@ -777,3 +777,54 @@ def test_copy_X_with_auto_gram():
     linear_model.lars_path(X, y, Gram='auto', copy_X=True, method='lasso')
     # X did not change
     assert_allclose(X, X_before)
+
+
+@pytest.mark.parametrize("LARS, has_coef_path, args",
+                         ((Lars, True, {}),
+                          (LassoLars, True, {}),
+                          (LassoLarsIC, False, {}),
+                          (LarsCV, True, {}),
+                          # max_iter=5 is for avoiding ConvergenceWarning
+                          (LassoLarsCV, True, {"max_iter": 5})))
+@pytest.mark.parametrize("x_data_type, y_data_type, expected_type", (
+    (np.float32, np.float32, np.float32),
+    (np.float64, np.float64, np.float64)))
+def test_lars_dtype_match(LARS, has_coef_path, args,
+                          x_data_type, y_data_type, expected_type):
+    rng = np.random.RandomState(0)
+    X = rng.rand(6, 6).astype(x_data_type)
+    y = rng.rand(6).astype(y_data_type)
+
+    model = LARS(**args)
+    model.fit(X, y)
+    assert model.coef_.dtype == expected_type
+    if has_coef_path:
+        assert model.coef_path_.dtype == expected_type
+    assert model.intercept_.dtype == expected_type
+
+
+@pytest.mark.parametrize("LARS, has_coef_path, args",
+                         ((Lars, True, {}),
+                          (LassoLars, True, {}),
+                          (LassoLarsIC, False, {}),
+                          (LarsCV, True, {}),
+                          # max_iter=5 is for avoiding ConvergenceWarning
+                          (LassoLarsCV, True, {"max_iter": 5})))
+def test_lars_numeric_consistency(LARS, has_coef_path, args):
```

Comment:
Please add a small comment to decribe the goal of this test

Reply:
Done. Thanks